src/Behat/Testwork/Tester/Cli/StrictController.php

```diff
@@ -29,21 +29,16 @@ final class StrictController implements Controller
      * @var ResultInterpreter
      */
     private $resultInterpreter;
-    /**
-     * @var bool
-     */
-    private $strict;
 
     /**
      * Initializes controller.
      *
      * @param ResultInterpreter $resultInterpreter
      * @param bool           $strict
      */
-    public function __construct(ResultInterpreter $resultInterpreter, $strict = false)
+    public function __construct(ResultInterpreter $resultInterpreter)
     {
         $this->resultInterpreter = $resultInterpreter;
-        $this->strict = $strict;
```

Comment:
I removed the "strict" injection, and moved the implementation in the TesterExtension (with a service that is tagged with `RESULT_INTERPRETATION_TAG`), which seems to be a better implementation. This way we avoid having core logic into the command domain.

The `ExerciseController` (that handle the `dry-run`) is not now migrated though. It can be done later if you think that it's a good vision. The other possibility is to let the Controller code "as-is", but in that case I would suggest to move the StopOnFailureController into the testwork namespace and remove the StopOnFailureHandler then (and keep the implementation in the controller).

Reply:
I'm not sure about this.

If it was new code, I prefer it the way you have done it. But it introduces another BC break - for example if someone is decorating/wrapping the StrictController and expecting it to respect the `$strict` parameter.

IMO keeping the controller as-is is not a huge downside for the benefit of minimising the risk of breaking someone's existing usage.

I agree for purity then it would make sense to rework the StopOnFailureController / StopOnFailureHandler to follow the same pattern - although I could also live with those being inconsistent if we made a note to reapply these changes for the next major.

Reply:
OK, I rolledback the changes in the StrictController.
In the end, I just added a test for this feature 😄